components/automate-ui/src/app/entities/layout/layout-sidebar.service.ts

```diff
@@ -1,286 +1,295 @@
-import { Injectable } from '@angular/core';
+import { Injectable, OnInit, OnDestroy } from '@angular/core';
 import { Store } from '@ngrx/store';
-import { Observable } from 'rxjs';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
 
 import { NgrxStateAtom } from 'app/ngrx.reducers';
 import { isProductDeployed } from 'app/staticConfig';
 import { isIAMv2 } from 'app/entities/policies/policy.selectors';
 import { FeatureFlagsService } from 'app/services/feature-flags/feature-flags.service';
 import { clientRunsWorkflowEnabled } from 'app/entities/client-runs/client-runs.selectors';
 import * as fromClientRuns from 'app/entities/client-runs/client-runs.reducer';
-
-import { MenuItemGroup } from './layout.model';
+import { UpdateSidebars } from './layout.actions';
 
 @Injectable({
     providedIn: 'root'
 })
-export class LayoutSidebarService {
+export class LayoutSidebarService implements OnInit, OnDestroy {
     public applicationsFeatureFlagOn: boolean;
-    public isIAMv2$: Observable<boolean>;
-    public ServiceNowfeatureFlagOn: boolean;
+    public isIAMv2: boolean;
+    public ServiceNowFeatureFlagOn: boolean;
+    private activeSidebar: string;
     private workflowEnabled: boolean;
-
+    private isDestroyed: Subject<boolean> = new Subject<boolean>();
 
     constructor(
         private store: Store<NgrxStateAtom>,
         private clientRunsStore: Store<fromClientRuns.ClientRunsEntityState>,
         private featureFlagsService: FeatureFlagsService
     ) {
-        this.isIAMv2$ = this.store.select(isIAMv2);
         this.applicationsFeatureFlagOn = this.featureFlagsService.getFeatureStatus('applications');
-        this.ServiceNowfeatureFlagOn = this.featureFlagsService.getFeatureStatus('servicenow_cmdb');
+        this.ServiceNowFeatureFlagOn = this.featureFlagsService.getFeatureStatus('servicenow_cmdb');
+        this.store.select(isIAMv2).pipe(
+          takeUntil(this.isDestroyed)
+        ).subscribe(
+            (IAMv2) => {
+                this.isIAMv2 = IAMv2;
+                this.updateSidebars();
```

Comment:
This appears to be wired up properly but it stops after this error occurs:
```
ERROR TypeError: You provided 'undefined' where a stream was expected. You can provide an Observable, Promise, Array, or Iterable.
    at subscribeTo (subscribeTo.js:27)
    at subscribeToResult (subscribeToResult.js:11)
    at SwitchMapSubscriber._innerSub (switchMap.js:46)
    at SwitchMapSubscriber._next (switchMap.js:36)
    at SwitchMapSubscriber.next (Subscriber.js:49)
    at WithLatestFromSubscriber._next (withLatestFrom.js:57)
    at WithLatestFromSubscriber.next (Subscriber.js:49)
    at WithLatestFromSubscriber._next (withLatestFrom.js:57)
    at WithLatestFromSubscriber.next (Subscriber.js:49)
    at FilterSubscriber._next (filter.js:33)
```
Unfortunately, that error trace is not very helpful in identifying where the error stems from. 🤷‍♂ 

To reproduce:
1. Change this line to 10 seconds for testing purpose: https://github.com/chef/automate/blob/70df1da94a0d8e40da232c4ca6e4b1382de57a84/components/automate-ui/src/app/entities/policies/policy.effects.ts#L53
2. rebuild automate-ui
3. Start with `chef-automate iam reset-to-v1`
4.  with settings page open, refresh browser and open dev tools (<kbd>cmd</kbd> + <kbd>opt</kbd> + <kbd>i</kbd>).
5. upgrade with `chef-automate iam upgrade-to-v2 --skip-policy-migration --beta2.1`
6. observe brower updates no more than 10 seconds later, **adding** "ACCESS MANAGEMENT" group in sidebar.
7. Downgrade again with `chef-automate iam reset-to-v1`
8. observe brower updates no more than 10 seconds later, **removing** "ACCESS MANAGEMENT" group in sidebar.
9. Also observe above error trace in browser console.


**Any further upgrades/downgrades will not update the sidebar.**

Reply:
@msorens if you could check if you still have the same issue. 👍🏻